### PR TITLE
Enable csi snapshotter

### LINKED ISSUE #457: Update deployments to include new snapshot controller (for v1beta1 APIs), start running tests, and update docs
Looks like some new snapshot APIs went Beta in k8s `1.17`. The corresponding `external-snapshotter` is also available and tested with the PD Driver. However, the stable deployment in this repo still does not contain the snapshotter, tests are not run by default, and the documentation for snapshotting may be out of date.

### PR DESCRIPTION
**Pull request by saikat-royc**
**What type of PR is this?**
/kind feature
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind flake

**What this PR does / why we need it**:
This PR enables csi snapshotter side car to GCE PD CSI driver.

**Which issue(s) this PR fixes**:

Fixes #457 

**Special notes for your reviewer**:
This is part 1 of the change. This change only includes yaml changes.
The second PR would add option to include SnapshotClassFile in https://github.com/saikat-royc/gcp-compute-persistent-disk-csi-driver/blob/master/test/k8s-integration/driver-config.go#L11 and add SnapshotClass tag to the template file (test-config-template.in).

**Does this PR introduce a user-facing change?**:

```release-note
Enable CSI Snapshotter (2.x) side car for PD CSI driver
```

### CODE REVIEW

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/controller_add_snapshotter.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+apiVersion: apps/v1
+metadata:
+  name: csi-gce-pd-controller
+spec:
+  template:
+    spec:
+      containers:
+        - name: csi-snapshotter
+          imagePullPolicy: Always
+          image: quay.io/k8scsi/csi-snapshotter:v2.1.0
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/controller_add_snapshotter.yaml:11]: this should be `gke.gcr.io/csi-snapshotter:v2.1.0-gke.0`

Comment (msau42) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/controller_add_snapshotter.yaml:11]: sorry, `gcr.io/gke-release-staging`

Comment (saikat-royc) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/controller_add_snapshotter.yaml]: Done

**Comment by msau42** [comment]
check cla

**Comment by msau42** [comment]
Can you also add a release note in the first comment?

File: deploy/kubernetes/overlays/prow-gke-release-staging-head/controller_add_snapshotter.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+apiVersion: apps/v1
+metadata:
+  name: csi-gce-pd-controller
+spec:
+  template:
+    spec:
+      containers:
+        - name: csi-snapshotter
+          imagePullPolicy: Always
+          image: quay.io/k8scsi/csi-snapshotter:2.1-canary
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-head/controller_add_snapshotter.yaml:11]: We want "canary" here to test latest changes in the master branch

Comment (saikat-royc) [reply deploy/kubernetes/overlays/prow-gke-release-staging-head/controller_add_snapshotter.yaml]: Done

**Pull request merged**